api/apps/geoprocessing/src/marxan-sandboxed-runner/adapters-blm/blm-partial-results.repository.ts

```diff
@@ -1,37 +1,61 @@
 import { Injectable } from '@nestjs/common';
-import { SandboxRunnerOutputHandler } from '../sandbox-runner-output-handler';
 import { Workspace } from '../ports/workspace';
-
-type SomePartialResults = {};
+import { existsSync, promises } from 'fs';
+import { MarxanOutputParserService } from '../adapters-shared/marxan-output-parser/marxan-output-parser.service';
+import { InjectRepository } from '@nestjs/typeorm';
+import { BlmPartialResultEntity } from './blm-partial-results.geo.entity';
+import { Repository } from 'typeorm';
+import { BestSolutionService } from '../adapters-shared/marxan-output-parser/best-solution.service';
+import { MarxanDirectory } from '../adapters-single/marxan-directory.service';
 
 @Injectable()
-export class BlmPartialResultsRepository
-  implements SandboxRunnerOutputHandler<SomePartialResults> {
+export class BlmPartialResultsRepository {
+  constructor(
+    private readonly marxanOutputParser: MarxanOutputParserService,
+    private readonly bestSolutionsService: BestSolutionService,
+    @InjectRepository(BlmPartialResultEntity)
+    private readonly repository: Repository<BlmPartialResultEntity>,
+    private readonly marxanDirectory: MarxanDirectory,
+  ) {}
+
   cancel(): Promise<void> {
     return Promise.resolve(undefined);
   }
 
-  /**
-   * save partial results (for given blm value)
-   * to some temporary table
-   *
-   * would be used within BlmRunnerService, passed down to RunnerService
-   */
-  async dump(
-    workspace: Workspace,
-    scenarioId: string,
-    stdOutput: string[],
-    stdErr: string[] | undefined,
-  ): Promise<SomePartialResults> {
-    return {};
-  }
-
-  dumpFailure(
+  async savePartialResult(
     workspace: Workspace,
     scenarioId: string,
-    stdOutput: string[],
-    stdError: string[],
+    blmValue: number,
   ): Promise<void> {
-    return Promise.resolve(undefined);
+    const { fullPath: fullOutputPath } = this.marxanDirectory.get(
+      'OUTPUTDIR',
+      workspace.workingDirectory,
+    );
+
+    if (!existsSync(fullOutputPath + `/output_sum.csv`)) {
+      throw new Error(`Output is missing from the marxan run.`);
+    }
+
+    const runsSummary = (
+      await promises.readFile(fullOutputPath + `/output_sum.csv`)
+    ).toString();
+
+    const runs = this.bestSolutionsService.map(
+      this.marxanOutputParser.parse(runsSummary),
+    );
+    const bestRun = runs.find((run) => run.best);
+
+    if (!bestRun) {
+      throw new Error(
+        `${scenarioId} scenario calibration doesn't have a best run for ${blmValue} BLM value`,
+      );
+    }
```

Comment:
This part could be a standalone class to make it easier to test and separate concerns.

Reply:
Maybe I can create `BestRunService`, what do you think?

Reply:
@aciddaute 👍 (be aware of the class names clashing tho - NestJS DI system uses class names)